test/graph_generation_tests.cc

```diff
@@ -1459,5 +1459,87 @@ namespace detail {
 		maybe_print_graphs(ctx);
 	}
 
+
+	TEST_CASE("buffer accesses with empty ranges do not generate pushes or data-flow dependencies", "[graph_generator][command-graph]") {
```

Comment:
I guess there is some overlap here with `"graph_generator allows chunks to require empty buffer ranges"`, but that's fine.

Reply:
You're right, that's basically a duplicate. I wasn't aware this was explicitly supported before at all. We even ran into the same issues with the range mappers while writing these tests!
```c++
// NOTE: It's important to construct range-mappers in such a way that passing the
// global size (during tdag generation) still returns the correct result!
```
Should we get rid of one of them, or are tests "the more the merrier"?

Reply:
Hmm I think having multiple tests is probably only confusing in the long run, so I'd say remove one of them. I'm fine with going with your new version, although I will say that I found the range mapper definition somewhat hard to understand (without running it).

Reply:
I've removed the old test, replaced one of my RMs with `one_to_one` and made the other one more readable.